modules/abstract-lightning/src/wallet/lightning.ts

```diff
@@ -312,6 +338,54 @@ export class LightningWallet implements ILightningWallet {
     };
   }
 
+  /**
+   * Private method to handle LNURL payments
+   * @param {SubmitPaymentParams} params - LNURL payment parameters
+   * @returns {Promise<PayInvoiceResponse>} Payment result
+   * @private
+   */
+  private async payLnurl(
+    params: t.TypeOf<typeof SubmitPaymentParams> & { lnurl: string }
+  ): Promise<PayInvoiceResponse> {
+    try {
+      if (!lnurlProcessor.isValidLnurl(params.lnurl)) {
+        throw new Error('Invalid LNURL format');
+      }
+      const amountSats = params.amountMsat / BigInt(1000);
+      const lnurlPaymentResult = await lnurlProcessor.processLnurlPayment(params.lnurl, amountSats, params.comment);
+      const comment = params.comment || lnurlPaymentResult.description;
+
+      // Create a new params object with the invoice instead of lnurl
+      return this.payInvoice({
+        ...params,
+        invoice: lnurlPaymentResult.invoice,
+        comment,
+        lnurlData: {
+          description: lnurlPaymentResult.description,
+          successAction: lnurlPaymentResult.successAction,
+        },
```

Comment:
```suggestion
        lnurlData: {
          description: lnurlPaymentResult.description,
          successAction: lnurlPaymentResult.successAction,
        },
```

What kind of data is passed here? not sure if we need this in the backend, we won't use it anyway?